backends/backends.go

```diff
@@ -37,9 +41,39 @@ func GetBackends(config *util.Config, timeout time.Duration) (map[string][]util.
 	}
 }
 
-func BackendManager(config *util.Config, forwarder *forward.Forwarder, updateChan chan time.Time) error {
-	consecutive_errors := 0
-	for {
+func BackendManager(config *util.Config, forwarder *forward.Forwarder, updateChan chan util.Update) (err error) {
+	go func() {
+		for {
+			select {
+			case update := <-updateChan:
+				delay := time.Now().Sub(update.Time)
+				log.Debug("Update requested",
+					zap.String("delay", delay.String()),
+					zap.String("reason", update.Reason),
+				)
+				trigger(update)
+			case <-time.After(time.Second * time.Duration(config.MinUpdateInterval)):
+				log.Debug("Min update interval reached without updates, forcing reload")
+				trigger(util.NewUpdate("min-update-interval-reached"))
+			}
+		}
+	}()
+
+	// watch changes in kubernetes api and trigger update
+	if !config.DisableWatch {
+		go kubernetes.WatchBackends(config, updateChan)
+	} else {
+		log.Info("API watch has been disabled by config flag, reloading changes with fixed interval only",
+			zap.Int("interval", config.MinUpdateInterval),
+		)
+	}
+
+	poll(func(update util.Update) {
+		log.Info("Reloading",
+			zap.String("time", update.Time.String()),
```

Comment:
the format of this is really wonky and can we please make it sane?
`"time":"2019-07-30 16:37:30.015862938 +0200 CEST m=+18.520433168"`

Reply:
(and rename the field to something other than "time" -- it's confusing since we already have "@timestamp")

Reply:
> can we please make it sane?

... that's an opinionated statement. Can you please clarify what you want? Something like rfc3339 or iso8601 -> `2019-07-31T09:37:13+00:00` ?

Reply:
Question is: Would you prefer "delay" here instead? Like in the log-statement in line ~50 which has now become Debug?